api/app/src/main/kotlin/packit/controllers/RoleController.kt

```diff
@@ -24,7 +24,7 @@ class RoleController(private val roleService: RoleService, private val userRoleS
     {
         val role = roleService.createRole(createRole)
 
-        return ResponseEntity.created(URI.create("/roles/${role.name}")).body(role.toDto())
+        return ResponseEntity.created(URI.create("/roles/${role.id}")).body(role.toDto())
```

Comment:
I'm confused by this. The `getRoleByName` endpoint requires a name, not a numerical ID. How does this fix anything? Why do spaces actually cause an issue? Do we need to add a role "slug"? Should we ban spaces from role names?

Reply:
I think what you want is something like:
```
UriComponentsBuilder.fromUriString("/roles/{}").build(role.name)
```

That will properly encode the argument into the URI string. Spaces in the name should become `%20` and slashes should become `%2F` (and we should make sure requests to `/roles/foo%2fbar` actually works the way we want it to).

Or we could just not return any URI here. I don't think the frontend uses it anyway.

Reply:
ohh yeah i thought we had /role/{id} endpoint... but you are right and we don't use URL anways so have removed... also good point with BE validation too so have added that